### PR TITLE
Equipment interaction lamps, laptop and solar panel

### PR DESCRIPTION
**Pull request by y0014984**
Rework of most interaction functions for the rugged equipment. Additionally some minor changes.

- adjustable lights (extend, yaw, pitch)
- ACE3 dragging/carrying workaround
- laptop open/close animation with power management
- table open/close animation
- changed light module into interaction module
- solar panel animations

### CODE REVIEW

File: addons/interaction/functions/fnc_laptop_addActionClose.sqf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Opens a laptop
```
Comment (GermanHydrogen) [inline addons/interaction/functions/fnc_laptop_addActionClose.sqf:2]: Why is this laptop specific?

File: addons/interaction/functions/fnc_laptop_addActionOpen.sqf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (GermanHydrogen) [inline addons/interaction/functions/fnc_laptop_addActionOpen.sqf:1]: Why is this laptop specific.

File: addons/interaction/functions/workaroundLight4.sqf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (GermanHydrogen) [inline addons/interaction/functions/workaroundLight4.sqf:1]: Depricated?

File: addons/armaos/CfgVehicles.hpp [hunk 4]
```diff
@@ -20,6 +20,18 @@ class CfgVehicles
 			//init = "params ['_entity']; call compile preprocessFileLineNumbers '\z\ae3\addons\main\init\initLaptop.sqf';";
 		};
 
+		class AE3_Equipment
```
Comment (GermanHydrogen) [inline addons/armaos/CfgVehicles.hpp:23]: Add interaction module to requiredAddons

**Review by GermanHydrogen — CHANGES_REQUESTED** [review]

File: addons/power/CfgVehicles.hpp [hunk 5]
```diff
@@ -129,4 +132,78 @@ class CfgVehicles
...
+						animation = "Panel_2_Pitch";
+						minValue = -90;
+						maxValue = 90;
+						scrollMultiplier = 10;
+					};
+				};
+				
+				class AE3_Animation_Point_2
+				{
+					description = "panels";
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/power/CfgVehicles.hpp:192]: Changed to "solar panel" in next commit.

File: addons/power/CfgVehicles.hpp [hunk 6]
```diff
@@ -129,4 +132,78 @@ class CfgVehicles
...
+						animation = "Panel_1_Pitch";
+						minValue = -90;
+						maxValue = 90;
+						scrollMultiplier = 10;
+					};
+				};
+
+				class AE3_Animation_Point_1
+				{
+					description = "panel 2";
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/power/CfgVehicles.hpp:177]: Changed to "solar panel" in next commit.

File: addons/power/CfgVehicles.hpp [hunk 7]
```diff
@@ -129,4 +132,78 @@ class CfgVehicles
...
+		{
+			displayName = "Solar Panel";
+
+			init = "_this call AE3_interaction_fnc_initSolarPanel;";
+
+			class AE3_Animations
+			{
+				class AE3_Animation_Point_0
+				{
+					description = "panel 1";
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/power/CfgVehicles.hpp:162]: Changed to "solar panel" in next commit.

File: addons/armaos/functions/fnc_computer_addActionTurnOff.sqf [hunk 8]
```diff
@@ -9,7 +9,7 @@
  * None
  */
 
-params ["_computer", "_silent"];
+params ["_computer", ["_silent", false]];
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/armaos/functions/fnc_computer_addActionTurnOff.sqf:12]: done in next commit.

File: addons/interaction/functions/fnc_initLaptop.sqf [hunk 9]
```diff
@@ -0,0 +1,20 @@
+/**
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/interaction/functions/fnc_initLaptop.sqf:1]: Perhaps there will be more laptop specific things in the future.

File: addons/interaction/functions/fnc_initLaptop.sqf [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+ * Initializes a laptop.
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/interaction/functions/fnc_initLaptop.sqf:2]: Added more information.

File: addons/interaction/functions/fnc_initSolarPanel.sqf [hunk 11]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Initializes a solar panel.
```
[the remark this answers is not in the document]
Comment (y0014984) [reply addons/interaction/functions/fnc_initSolarPanel.sqf:2]: This one is especially for you @GermanHydrogen . I thought you would need it if you do the calculation stuff for the solar panel power production. I simply used the same systematic for all asset types.

File: addons/interaction/functions/fnc_laptop_addActionClose.sqf (hunk 1 above)
Comment (y0014984) [reply addons/interaction/functions/fnc_laptop_addActionClose.sqf]: Only the laptop has that specific open/close animation. The is already an abstraction layer in config, using the following asset neutral settings:

			openAction = 
			closeAction =

File: addons/armaos/CfgVehicles.hpp (hunk 4 above)
Comment (y0014984) [reply addons/armaos/CfgVehicles.hpp:23]: done

**Comment by y0014984** [comment]
My To-Do's before leaving draft state of this PR:

- testing on dedicated server
- move chair to interaction module
- modify ACE Interaction points for lamps

**Comment by y0014984** [comment]
Testing on dedicated server does not show any major problems. Animations, sound etc. work as expected. But there is one minor issue:
- Disabled dragging for open table only works on player opening the table

Additionally there are some issues not directly related to this PR:
- Power and Fuel Consumption doesn't seem to work on dedicated server
- Laptop is simultaniously useable
- Generator loop sound still plays if player A turns on the generator and player B turns off the generator
- ArmaOS says root user doesn't exist, if you try to login; there should be a login prohibited message

**Review by GermanHydrogen — APPROVED** [review]

**Pull request merged**